src/gtest/test_libzendoo.cpp

```diff
@@ -2491,3 +2500,409 @@ TEST(CctpLibrary, TestCustomFieldsValidationFix)
+
+    SelectParams(CBaseChainParams::REGTEST);
+    BlockchainTestManager::GetInstance().GenerateSidechainTestParameters(testProvingSystem, TestCircuitType::Certificate, false);
+    BlockchainTestManager::GetInstance().GenerateSidechainTestParameters(testProvingSystem, TestCircuitType::CSW, false);
+
+    uint256 sidechainId = uint256S("aabba");
+
+    CSidechain sidechain;
+    sidechain.creationBlockHeight = 100;
+    sidechain.fixedParams.withdrawalEpochLength = 20;
+    sidechain.fixedParams.constant = CFieldElement{SAMPLE_FIELD};
+    sidechain.fixedParams.version = 0;
+    sidechain.lastTopQualityCertHash = uint256S("cccc");
+    sidechain.lastTopQualityCertQuality = 100;
+    sidechain.lastTopQualityCertReferencedEpoch = -1;
+    sidechain.lastTopQualityCertBwtAmount = 50;
+    sidechain.balance = CAmount(100);
+    sidechain.fixedParams.wCertVk = BlockchainTestManager::GetInstance().GetTestVerificationKey(testProvingSystem, TestCircuitType::Certificate);
+    sidechain.fixedParams.wCeasedVk = BlockchainTestManager::GetInstance().GetTestVerificationKey(testProvingSystem, TestCircuitType::CSW);
+    
+    BlockchainTestManager& testManager = BlockchainTestManager::GetInstance();
+    testManager.Reset();
+
+    // Store the test sidechain and extend the blockchain to complete at least one epoch. 
+    testManager.StoreSidechainWithCurrentHeight(sidechainId, sidechain, sidechain.creationBlockHeight + sidechain.fixedParams.withdrawalEpochLength);
+
+    CTxCeasedSidechainWithdrawalInput csw = testManager.CreateCswInput(sidechainId, 1, testProvingSystem);
+
+    // FT and CSW in the same tx
+    SidechainTxsCommitmentBuilder cmtObj;
+    CMutableTransaction mtx, mtx2, mtx3;;
+    mtx.nVersion = SC_TX_VERSION;
+    mtx.vsc_ccout.resize(0);
+    mtx.vft_ccout.resize(0);
+    mtx.vmbtr_out.resize(0);
+    mtx.vcsw_ccin.resize(0);
+    mtx2.nVersion = SC_TX_VERSION;
+    mtx2.vsc_ccout.resize(0);
+    mtx2.vft_ccout.resize(0);
+    mtx2.vmbtr_out.resize(0);
+    mtx2.vcsw_ccin.resize(0);
+    mtx3.nVersion = SC_TX_VERSION;
+    mtx3.vsc_ccout.resize(0);
+    mtx3.vft_ccout.resize(0);
+    mtx3.vmbtr_out.resize(0);
+    mtx3.vcsw_ccin.resize(0);
+
+    // forward transfer
+    mtx.vft_ccout.push_back(CTxForwardTransferOut(sidechainId, CAmount(43), uint256S("abba101"), uint160S("abba101")));
+    // ceased sidechain withdrawals
+    mtx.vcsw_ccin.push_back(csw);
+
+    CTransaction tx(mtx);
+    ASSERT_FALSE(cmtObj.add(tx));
+
+
+    // forward transfer
+    mtx2.vft_ccout.push_back(CTxForwardTransferOut(sidechainId, CAmount(43), uint256S("abba101"), uint160S("abba101")));
+    CTransaction tx2(mtx2);
+
+    // ceased sidechain withdrawals
+    mtx3.vcsw_ccin.push_back(csw);
+    CTransaction tx3(mtx3);
+
+    // tx with CSW after tx with FT
+    SidechainTxsCommitmentBuilder cmtObj2;
+    ASSERT_TRUE(cmtObj2.add(tx2));
+    ASSERT_FALSE(cmtObj2.add(tx3));
+
+    // tx with CSW after tx with FT
```

Comment:
Is it possible that the comment here should be reversed?

```suggestion
    // tx with FT after tx with CSW
```

Reply:
Correct. Addressed in 9ce2a8aff3